jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -788,7 +792,158 @@ protected void cleanupAfterStop() throws Exception
 
         // if we're not persisting the temp dir contents delete it
         if (tempDirectory != null && tempDirectory.exists() && !isTempDirectoryPersistent())
+        {
             IO.delete(tempDirectory);
+        }
+
+        //if it was jetty that created the tmp dir, it can be reset, otherwise we need to retain the name
+        if (_createdTempDirectoryName)
+        {
+            setTempDirectory(null);
+            _createdTempDirectoryName = false;
+        }
+    }
+
+    protected void makeTempDirectory()
```

Comment:
Need javadoc to say what this is and how it differs from setTempDirector

Reply:
Done.